conda/models/environment.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""EXPERIMENTAL Conda environment data model"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from logging import getLogger
+from typing import TYPE_CHECKING
+
+from ..exceptions import CondaValueError
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from .match_spec import MatchSpec
+    from .records import PackageRecord
+
+
+log = getLogger(__name__)
+
+
+@dataclass
+class Environment:
+    """
+    **Beta** While in beta, expect both major and minor changes across minor releases.
```

Comment:
Marking as in beta, following the pattern defined in the [api module](https://github.com/conda/conda/blob/main/conda/api.py).